http4s/src/main/scala/pl/abankowski/httpsigner/http4s/impl/package.scala

```diff
@@ -21,15 +21,15 @@ package object impl {
   }
 
   trait RequestHelpers[F[_]] extends Helpers{
-    implicit val F: Sync[F]
+    implicit val F: Async[F]
     val config: HttpCryptoConfig
 
     protected def message(request: Request[F]): F[Array[Byte]] =
       F.delay {
-        (List(request.method.name, request.uri.path, request.uri.query.renderString) ++
-          request.headers.toList.collect({
+        (List(request.method.name, request.uri.path.toString(), request.uri.query.renderString) ++
+          request.headers.headers.collect({
```

Comment:
It's odd that in http4s they used path `request.headers.headers`.

Reply:
Yeah. `Request` has a field `val headers: Headers` and `Headers` is a newtype:
```final class Headers(val headers: List[Header.Raw]) extends AnyVal```